### PR TITLE
Create statesync.bash

### PR DESCRIPTION
**Pull request by faddat**
state sync juno with pebble

### CODE REVIEW

**Review by the-frey — DISMISSED** [review]

File: scripts/statesync.bash [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+mv juno-phoenix2-genesis.json $HOME/.juno/config/genesis.json
+
+
+
+
+# Get "trust_hash" and "trust_height".
+INTERVAL=1000
+LATEST_HEIGHT=$(curl -s https://juno-rpc.polkachu.com/block | jq -r .result.block.header.height)
+BLOCK_HEIGHT=$(($LATEST_HEIGHT-$INTERVAL)) 
+TRUST_HASH=$(curl -s "https://juno-rpc.polkachu.com/block?height=$BLOCK_HEIGHT" | jq -r .result.block_id.hash)
```
Comment (joeabbey) [inline scripts/statesync.bash:35]:
```suggestion
# Get "trust_hash" and "trust_height".
INTERVAL=1000
LATEST_HEIGHT="$(curl -s https://juno-rpc.polkachu.com/block | jq -r .result.block.header.height)"
BLOCK_HEIGHT="$(($LATEST_HEIGHT-$INTERVAL))"
TRUST_HASH="$(curl -s "https://juno-rpc.polkachu.com/block?height=$BLOCK_HEIGHT" | jq -r .result.block_id.hash)"
```

File: scripts/statesync.bash [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+# Export state sync variables.
+export JUNOD_STATESYNC_ENABLE=true
+export JUNOD_P2P_MAX_NUM_OUTBOUND_PEERS=200
+export JUNOD_STATESYNC_RPC_SERVERS="https://rpc-juno-ia.notional.ventures:443,https://juno-rpc.polkachu.com:443"
+export JUNOD_STATESYNC_TRUST_HEIGHT=$BLOCK_HEIGHT
+export JUNOD_STATESYNC_TRUST_HASH=$TRUST_HASH
+
+# Fetch and set list of seeds from chain registry.
+export JUNOD_P2P_SEEDS=$(curl -s https://raw.githubusercontent.com/cosmos/chain-registry/master/juno/chain.json | jq -r '[foreach .peers.seeds[] as $item (""; "\($item.id)@\($item.address)")] | join(",")')
```
Comment (joeabbey) [inline scripts/statesync.bash:49]:
```suggestion
export JUNOD_P2P_SEEDS="$(curl -s https://raw.githubusercontent.com/cosmos/chain-registry/master/juno/chain.json | jq -r '[foreach .peers.seeds[] as $item (""; "\($item.id)@\($item.address)")] | join(",")')"
```

**Comment by joeabbey** [comment]
Overall looks good .. just a couple bash globbing nits

**Comment by faddat** [comment]
what do the quotes do?

**Comment by joeabbey** [comment]
> what do the quotes do?

Prevents accidental word splitting:
https://github.com/koalaman/shellcheck/wiki/SC2046

**Comment by faddat** [comment]
thanks!

File: scripts/statesync.bash [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+# go install -ldflags '-w -s -X github.com/cosmos/cosmos-sdk/types.DBBackend=badgerdb' -tags badgerdb ./...
+# go install -ldflags '-w -s -X github.com/cosmos/cosmos-sdk/types.DBBackend=boltdb' -tags boltdb ./...
+
+# Initialize chain.
+junod init test
+
+# Get Genesis
+wget https://download.dimi.sh/juno-phoenix2-genesis.tar.gz
+tar -xvf juno-phoenix2-genesis.tar.gz
+mv juno-phoenix2-genesis.json $HOME/.juno/config/genesis.json
```
Comment (joeabbey) [inline scripts/statesync.bash:26]:
```suggestion
mv juno-phoenix2-genesis.json "$HOME/.juno/config/genesis.json"
```

File: scripts/statesync.bash [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+
+
+
+# Get "trust_hash" and "trust_height".
+INTERVAL=1000
+LATEST_HEIGHT="$(curl -s https://juno-rpc.polkachu.com/block | jq -r .result.block.header.height)"
+BLOCK_HEIGHT="$(($LATEST_HEIGHT-$INTERVAL))"
```
Comment (joeabbey) [inline scripts/statesync.bash:34]:
```suggestion
BLOCK_HEIGHT="$((LATEST_HEIGHT-INTERVAL))"
```

File: scripts/statesync.bash [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+# Export state sync variables.
+export JUNOD_STATESYNC_ENABLE=true
+export JUNOD_P2P_MAX_NUM_OUTBOUND_PEERS=200
+export JUNOD_STATESYNC_RPC_SERVERS="https://rpc-juno-ia.notional.ventures:443,https://juno-rpc.polkachu.com:443"
+export JUNOD_STATESYNC_TRUST_HEIGHT=$BLOCK_HEIGHT
+export JUNOD_STATESYNC_TRUST_HASH=$TRUST_HASH
+
+# Fetch and set list of seeds from chain registry.
+export JUNOD_P2P_SEEDS="$(curl -s https://raw.githubusercontent.com/cosmos/chain-registry/master/juno/chain.json | jq -r '[foreach .peers.seeds[] as $item (""; "\($item.id)@\($item.address)")] | join(",")')"
```
Comment (joeabbey) [inline scripts/statesync.bash:49]:
```suggestion
JUNOD_P2P_SEEDS="$(curl -s https://raw.githubusercontent.com/cosmos/chain-registry/master/juno/chain.json | jq -r '[foreach .peers.seeds[] as $item (""; "\($item.id)@\($item.address)")] | join(",")')"
export JUNOD_P2P_SEEDS
```

**Comment by joeabbey** [comment]
> thanks!

just passed it through shellcheck and it flagged those last three

**Comment by faddat** [comment]
this is the way

**Review by joeabbey — APPROVED** [review]

**Review by JakeHartnell — APPROVED** [review]

**Comment by sascha1337** [comment]
+5000 🚀 very helpful, straight gonna test it

**Pull request merged**